internal/data-ingest-cli/receiver/otlphttp/otlphttp.go

```diff
@@ -0,0 +1,191 @@
+	case <-time.After(10 * time.Second):
+	}
+}
+
+func (r *OTLPHTTPReceiver) handleTraces(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		err := fmt.Errorf("Invalid request method %s", req.Method)
+		http.Error(w, err.Error(), http.StatusMethodNotAllowed)
+		log.Fatalln(err)
+		return
+	}
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		err := fmt.Errorf("Failed to read request body %s", err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Fatalln(err)
+		return
+	}
+
+	defer req.Body.Close()
+
+	unmarshaler := ptrace.ProtoUnmarshaler{}
+	traces, err := unmarshaler.UnmarshalTraces(body)
+	if err != nil {
+		log.Println("Failed to unmarshal traces to proto, checking JSON...")
+		unmarshaler := ptrace.JSONUnmarshaler{}
+		traces, err = unmarshaler.UnmarshalTraces(body)
+		if err != nil {
+			http.Error(w, "Failed to unmarshal traces", http.StatusBadRequest)
+			log.Fatalln("Failed to unmarshal traces to JSON and proto")
+			return
+		}
+	}
+
+	tracesMarshaler := &ptrace.JSONMarshaler{}
+	data, err := tracesMarshaler.MarshalTraces(traces)
+	if err != nil {
+		err := fmt.Errorf("Failed to marshal traces to JSON %s", err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Fatalln(err)
+		return
+	}
+
+	receiver.WriteToFile(r.config.OutputFile, data)
+	r.receivedDataChan <- struct{}{}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (r *OTLPHTTPReceiver) handleMetrics(w http.ResponseWriter, req *http.Request) {
+	log.Println("Received metrics")
+	if req.Method != http.MethodPost {
+		err := fmt.Errorf("Invalid request method %s", req.Method)
+		http.Error(w, err.Error(), http.StatusMethodNotAllowed)
+		log.Fatalln(err)
+		return
+	}
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		err := fmt.Errorf("Failed to read request body %s", err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Fatalln(err)
+		return
+	}
+
+	defer req.Body.Close()
+
+	unmarshaler := pmetric.ProtoUnmarshaler{}
+	metrics, err := unmarshaler.UnmarshalMetrics(body)
+	if err != nil {
+		log.Println("Failed to unmarshal metrics to proto, checking JSON...")
+		unmarshaler := pmetric.JSONUnmarshaler{}
+		metrics, err = unmarshaler.UnmarshalMetrics(body)
+		if err != nil {
+			http.Error(w, "Failed to unmarshal metrics", http.StatusBadRequest)
+			log.Fatalln("Failed to unmarshal metrics to JSON and proto")
+			return
+		}
+	}
+
+	metricsMarshaler := &pmetric.JSONMarshaler{}
+	data, err := metricsMarshaler.MarshalMetrics(metrics)
+	if err != nil {
+		err := fmt.Errorf("Failed to marshal metrics to JSON %s", err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Fatalln(err)
+		return
+	}
+
+	receiver.WriteToFile(r.config.OutputFile, data)
+	r.receivedDataChan <- struct{}{}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (r *OTLPHTTPReceiver) handleLogs(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
```

Comment:
let's extract this piece into its own function so we can keep the handler methods a bit shorter